### PR TITLE
fix: enforce resting-state component contracts

### LINKED ISSUE #931: Epic: tests assert the transition but not the resulting state, and nothing checks resting appearance
### Gap A: assertions prove a transition happened, not the state it left behind

- `command-menu.test.ts:178` is `test('Escape dismisses the menu')`, firing `keyDown(anchor, {key:'Escape'})` at `:188`, with another Escape case at `:336`. Both assert the menu **closed**. Neither asserts it **stays** closed — and it does not: `command-menu.svelte:147` wires `onEscape: dismiss`, but nothing clears the `/` trigger character, so `detectTrigger` immediately reopens the menu. Escape is effectively inert and the suite is green.
- `load-more.svelte` has the mirror-image gap. `rootMargin = '200px 0px'` (`:29`) means the sentinel is permanently intersecting in any short list. `handleIntersect` (`:108-110`) fires `requestNextPage('sentinel')`; guards at `:78` cover `!hasMore || loading || requestInFlight` — i.e. **concurrent** requests only. `requestInFlight` clears in a `finally` (`:104`), so each completed load immediately permits the next while the sentinel stays in view. It loads repeatedly until `retryCount >= maxRetries` stops it — and `retryCount`, despite its name, increments on **every** sentinel load (`:87-89`) and resets only on button loads (`:95-96`), so an error-retry budget is doubling as the auto-load cap. Tests assert it fires; nothing asserts it stops.

### Gap B: nothing checks a component's resting appearance

Two components shipped visibly broken with green suites:

- `sortable-list.css:13-17` — `.cinder-sortable-item { position: relative; cursor: default; user-select: none; }` is the complete base rule. **No `display`, no `align-items`, no `gap`.** Drag handles stack below their labels as block boxes. The drag states (`--lifted` `:19`, `--placeholder` `:38`, `--shifting` `:52`) are thoroughly styled — the interaction got attention, the resting layout got none.
- `logo-cloud.css` — `[data-cinder-columns='3']`, `='4'`, `='5'`, `='6']` **all emit `repeat(2, minmax(0,1fr))`**, identical to the default. The `columns` prop is inert. Shipped.

Also caught only by eye this session, not by any test: Card's `padding="none"` stripping header/footer padding so the title collides with the card edge; the Callout caret rendering wrong; SelectableRow's two-tone hover (see issue tracking the row/item family consolidation).

---

## Fix A — contract-test requirement (land first, cheap)

Any component with a dismissal, toggle, or auto-load behaviour must assert the **steady state** after the transition, not just the transition itself. Concretely:

- Escape/dismissal tests must assert the surface **stays** closed after the triggering condition persists (e.g. re-run the trigger detection tick and assert still-closed), not just that it closed once.
- Auto-load tests must assert `onloadmore` fires **once** with a sentinel that remains in view (i.e. simulate the sentinel staying intersecting across multiple ticks and assert no repeated/uncapped firing), not just that it fires at all.

Concrete fixes required as part of this issue, not deferred:

- [ ] `command-menu.svelte`: clear the `/` trigger character (or otherwise suppress `detectTrigger`) on Escape dismissal so the menu does not immediately reopen. Add a regression test in `command-menu.test.ts` asserting the menu stays closed after Escape with the trigger character still present in the input.
- [ ] `load-more.svelte`: decouple the auto-load cap from `retryCount` — introduce a distinct counter/guard for "no more auto-loads once N consecutive loads returned via sentinel with no user action," and stop conflating it with the error-retry budget. Add a regression test asserting `onloadmore` fires exactly once when the sentinel remains permanently intersecting and no new items arrive to move it out of view.

Extend the existing contract-test/coverage-map guardrails so this requirement is enforced rather than remembered:

- [ ] Audit the rest of the dismissal/toggle/auto-load tests in the suite (popovers, dropdowns, sheets, drawers, other infinite-scroll consumers) for the same incomplete shape — transition asserted, resting state not — and file/fix each incomplete case found.

---

## Fix B — resting-state visual baselines (higher setup cost, highest ongoing value)

Extend `docs/visual-regression/` to snapshot the resting state of **every** example. The surface already exists and must not be broken: `/page/<name>?snapshot=1` renders example mounts once each on a pure-white surface — this is the testing package's mount contract.

Sequence:

- [ ] Inventory current visual-regression coverage against `docs/visual-regression/phase-0-inventory.md`; identify every example currently missing a resting-state baseline.
- [ ] Generate baselines for all examples via the `snapshot=1` mount surface.
- [ ] Wire baseline capture + diff into the browser-tests workflow (the workflow that gates PRs per the existing CI gating model).
- [ ] Document the review/approval flow for intentional visual diffs in `docs/visual-regression/baselines.md` (who approves, how to regenerate, how a diff is distinguished from a regression).
- [ ] Confirm, retroactively, that resting-state baselines for `SortableList`, `LogoCloud`, `Card` (`padding="none"`), `Callout`, and `SelectableRow` would have failed pre-fix and pass post-fix, as a smoke test of the mechanism.

---

## Acceptance criteria

- [ ] `command-menu.svelte` Escape fix landed with a regression test proving steady-state (menu stays closed).
- [ ] `load-more.svelte` auto-load cap decoupled from `retryCount`, with a regression test proving it fires once and stops while the sentinel remains in view.
- [ ] Contract-test/coverage-map guardrail extended to require steady-state assertions for dismissal/toggle/auto-load behaviours; audit of existing dismissal tests completed with findings fixed or filed.
- [ ] Resting-state visual baselines exist for every example enumerated in `docs/visual-regression/phase-0-inventory.md`, wired into the browser-tests CI workflow, with the approval flow documented in `baselines.md`.
- [ ] The five caught-by-eye regressions (SortableList layout, LogoCloud columns, Card padding="none", Callout caret, SelectableRow hover) are confirmed fixed and covered by the new baselines.

## Verification

```bash
bun run --filter=@lostgradient/cinder validate
bun run test
bun run test:browser
```

---

Related epics: #919 (hand-rolled primitives — SortableList/LogoCloud resting-state bugs are exactly the class of defect a shared, tested base would catch), #921 (surface ramp), #922 (prop/API vocabulary), #923 (z-index scale). Fix B's baseline mechanism is also the natural home for verifying whatever visual consolidation comes out of the component-inventory epic.

### PR DESCRIPTION
**Pull request by stevekinney**
Closes #931

Harden transition contracts and resting-state visuals for the affected component examples:

- Keep CommandMenu dismissed while the host trigger text and caret remain unchanged after Escape.
- Track LoadMore sentinel auto-loads separately from error handling and stop repeated callbacks at the configured cap.
- Restore SortableList's resting flex layout and make LogoCloud column values map to their declared track counts.
- Keep Card header and footer padding when `padding="none"` is used.
- Add focused contract tests for these states and committed Docker-authored visual baselines for Card, Callout, CommandMenu, LoadMore, LogoCloud, SelectableRow, and SortableList examples.

Verification:
- Focused component suites: 117 passing
- `bun run --filter=@lostgradient/cinder typecheck`
- `bun run --filter=@lostgradient/cinder components:check`
- `bun run --filter=@lostgradient/cinder exports:check`
- Canonical browser-tests workflow `30166388289`: scope, update-baselines, typecheck, and Playwright all passed

### CODE REVIEW

File: packages/components/src/components/logo-cloud/logo-cloud.css [hunk 1]
```diff
@@ -36,19 +36,19 @@
...
   }
 
   .cinder-logo-cloud[data-cinder-columns='4'] .cinder-logo-cloud__list {
-    grid-template-columns: repeat(2, minmax(0, 1fr));
+    grid-template-columns: repeat(4, minmax(0, 1fr));
   }
 
   .cinder-logo-cloud[data-cinder-columns='5'] .cinder-logo-cloud__list {
-    grid-template-columns: repeat(2, minmax(0, 1fr));
+    grid-template-columns: repeat(5, minmax(0, 1fr));
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/logo-cloud/logo-cloud.css]: Restored LogoCloud narrow-container behavior: the base remains two columns, while the configured 3/4/5/6 counts apply only inside the existing 48rem and 64rem container queries. Updated the regression test to assert those responsive rules. Fixed in 6bbb8e6238b47a8b6edba6dfe6e07ac414681c16.

File: packages/components/src/components/card/card.test.ts [hunk 2]
```diff
@@ -27,6 +28,16 @@ const emptySnippet = createRawSnippet(() => ({
 }));
 
 describe('Card', () => {
+  test('keeps header and footer padding when body padding is none', () => {
+    const stylesheet = readFileSync(new URL('./card.css', import.meta.url), 'utf8');
+    expect(stylesheet).not.toContain(
+      ".cinder-card[data-cinder-padding='none'] > .cinder-card__header",
+    );
+    expect(stylesheet).not.toContain(
+      ".cinder-card[data-cinder-padding='none'] > .cinder-card__footer",
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/card/card.test.ts]: Restored the documented Card contract: padding="none" removes body, header, and footer padding. The CSS regression test now asserts those selectors remain present. Fixed in 6bbb8e6238b47a8b6edba6dfe6e07ac414681c16.

File: packages/components/src/components/command-menu/command-menu.svelte [hunk 3]
```diff
@@ -135,6 +162,9 @@
 
   function dismiss() {
     if (!open) return;
+    if (anchor) {
+      dismissedTrigger = { anchor, value: anchor.value, caretIndex };
+    }
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/command-menu/command-menu.svelte]: The persistent dismissed-trigger latch now applies only to Escape. Outside-pointer dismissal remains transient, so returning to the unchanged anchor can intentionally reopen the menu. Added behavior coverage through the existing dismissal tests. Fixed in 6bbb8e6238b47a8b6edba6dfe6e07ac414681c16.

File: packages/components/src/components/sortable-list/sortable-list.css [hunk 4]
```diff
@@ -11,6 +11,9 @@
   }
 
   .cinder-sortable-item {
+    display: flex;
+    align-items: center;
+    gap: var(--cinder-space-2);
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/sortable-list/sortable-list.css:17]: Scoped the resting flex layout to `.cinder-sortable-list .cinder-sortable-item`, so Kanban cards retain their grid layout while standalone SortableList rows get the handle-and-label alignment. Added/updated the CSS contract test. Fixed in 4882b143.

File: packages/components/src/components/command-menu/command-menu.svelte [hunk 5]
```diff
@@ -103,23 +102,29 @@
...
 
   $effect(() => {
-    commandList.syncListboxId(listboxId);
+    const currentAnchor = anchor;
+    const currentValue = currentAnchor?.value;
+    if (
+      dismissedTrigger &&
+      (dismissedTrigger.anchor !== currentAnchor ||
+        dismissedTrigger.value !== currentValue ||
+        dismissedTrigger.caretIndex !== caretIndex)
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/command-menu/command-menu.svelte:117]: Fixed in accbef513. The Escape latch now captures the anchor’s actual selectionStart/selectionEnd and an anchor-level listener remains active while the menu is closed, clearing the latch as soon as selection or value moves. The regression covers moving outside the trigger and returning to the original caret; the full focused CommandMenu file passes 21/21 tests.

File: packages/components/src/components/sortable-list/sortable-list.css [hunk 6]
```diff
@@ -10,7 +10,10 @@
     padding: 0;
   }
 
-  .cinder-sortable-item {
+  .cinder-sortable-list .cinder-sortable-item {
+    display: flex;
+    align-items: center;
+    gap: var(--cinder-space-2);
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/sortable-list/sortable-list.css]: Fixed in 6af654f54. The standalone SortableList root now carries a dedicated data-cinder-sortable-list boundary, and the resting row layout selector is a direct-child selector under that attribute. Kanban intentionally shares the generic sortable classes but not this standalone-root attribute, so its cards retain their grid/start layout. The regression failed on the shared-class selector and now passes; the full SortableList file is 55/55, with Prettier, Stylelint, Oxlint, and the official Svelte checker clean.

File: packages/components/src/components/card/card.css [hunk 7]
```diff
@@ -95,6 +95,11 @@
     padding: 0;
   }
 
+  .cinder-card[data-cinder-padding='none'] > .cinder-card__header,
+  .cinder-card[data-cinder-padding='none'] > .cinder-card__footer {
+    padding: 0;
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/card/card.css]: Fixed in 554467c23. The contradictory root-level padding selectors and their opposite assertion were removed. `padding="none"` again targets only `.cinder-card__body`, preserving the documented header/footer inset. The focused Card suite passes 22/22; the canonical Docker run supplied the regression evidence by observing header 0px instead of 12px before this fix.

File: packages/components/src/components/sortable-list/sortable-list.css [hunk 8]
```diff
@@ -10,7 +10,10 @@
     padding: 0;
   }
 
-  .cinder-sortable-item {
+  [data-cinder-sortable-list] > .cinder-sortable-item {
+    display: flex;
+    align-items: center;
+    gap: var(--cinder-space-2);
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/sortable-list/sortable-list.css]: Fixed in 407db1e15. Standalone SortableList preview portals receive a dedicated `data-cinder-sortable-list-preview` boundary only when the source row belongs to a standalone data-cinder-sortable-list root. The row-layout rule covers that boundary without matching Kanban previews. The regression failed before the marker/style and now passes; full SortableList validation is 56/56, with Prettier, Stylelint, Oxlint, and the official Svelte checker clean.

File: packages/testing/snapshots/provenance.json [hunk 9]
```diff
@@ -1,7 +1,15 @@
...
+  "componentScope": [
+    "callout",
+    "card",
+    "command-menu",
+    "load-more",
+    "logo-cloud",
+    "selectable-row",
+    "sortable-list"
+  ],
+  "renderedSourceSha": "7eed6f470e4f3f4dc59df230d756c273249bb93d",
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/testing/snapshots/provenance.json]: Resolved by canonical browser-tests run 30170818525 and bot commit eae813f07. The pinned Playwright 1.60.0 Jammy/x64 image rendered the final source at 407db1e15 with scope `card,logo-cloud`; Card produced no pixel delta after its body-only padding contract was restored, while the four affected mobile/tablet LogoCloud baselines changed. provenance.json now records exactly that scope and rendered source SHA. The guarded source-branch push succeeded only after confirming the branch still matched the rendered head.

File: packages/components/src/components/_sortable-item.svelte [hunk 10]
```diff
@@ -128,6 +128,9 @@
     portal.setAttribute('aria-hidden', 'true');
     portal.setAttribute('inert', '');
     portal.className = 'cinder-sortable-drag-preview';
+    if (rowEl.closest('[data-cinder-sortable-list]')) {
+      portal.setAttribute('data-cinder-sortable-list-preview', '');
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/_sortable-item.svelte]: Fixed in b92dd15b0. Preview layout tagging now requires the sortable row's direct parent to be the standalone SortableList root, so it cannot cross a nested component boundary into an outer SortableList. Added a regression that simulates an outer marker beyond the direct parent; the preview remains untagged. Focused SortableList suite passes 57/57.
